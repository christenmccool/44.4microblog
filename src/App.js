import './App.css';
import React, {useEffect, useState} from 'react';
import PostContext from './PostContext';
import NavBar from './NavBar';
import Routes from './Routes';
import { v4 as uuid } from 'uuid';

const App = () => {

  // const addComment = (postId, comment) => {
  //   setPosts(posts.map(post => {
  //     if (post.id === postId) return {...post, comments: [...post.comments, {comment, id: uuid()}]};
  //     return post;
  //   }))
  // }

  // const removeComment = (postId, commentid) => {
  //   setPosts(posts.map(post => {
  //     if (post.id === postId) return {...post, comments: post.comments.filter(comment => comment.id !== commentid)};
  //     return post;
  //   }))
  // }


  return (
    <div className="App">
      <NavBar />
      <Routes />
    </div>
  );
}

export default App;
