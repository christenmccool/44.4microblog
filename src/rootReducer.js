import {ADD_POST, DELETE_POST, EDIT_POST, ADD_COMMENT, DELETE_COMMENT} from './actionTypes';

const postsInStorage = JSON.parse(localStorage.getItem("microblog-posts")) || {};

const INITIAL_STATE = {posts: postsInStorage};

const rootReducer = (state=INITIAL_STATE, action) => {
  switch (action.type) {
    case ADD_POST: {
      return {...state, posts: {...state.posts, [action.id]: {...action.post} }}
    }

    case DELETE_POST: {
      const updatedPosts = {...state.posts};
      delete updatedPosts[action.id];
      return ({...state, posts: updatedPosts});
    }

    case EDIT_POST: {
      const updatedPosts = {...state.posts};
      updatedPosts[action.id] = action.post;
      return ({...state, posts: updatedPosts});
    }

    case ADD_COMMENT: {
      const post = state.posts[action.postId];
      console.log(post);
      return {...state, posts: {...state.posts, [action.postId]: 
                                                  {...post, comments: [...post.comments, action.comment]}
                                }
              }
    }

    case DELETE_COMMENT: {
      const post = state.posts[action.postId];
      const updatedComments = post.comments.filter(comment => comment.id !== action.commentId);
      return {...state, posts: {...state.posts, [action.postId]: 
                                                  {...post, comments: updatedComments}
                                }
              }
    }
    

    default: return state;
  }
} 

export default rootReducer;