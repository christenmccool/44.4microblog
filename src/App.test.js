import { render, screen } from '@testing-library/react';
import App from './App';
import {MemoryRouter} from 'react-router-dom';
import {createStore, applyMiddleware} from 'redux';
import {Provider} from 'react-redux';
import thunk from "redux-thunk";
import rootReducer from './rootReducer';


const store = createStore(rootReducer, applyMiddleware(thunk));

test('renders without crashing', () => {
  render( <Provider store={store}>
           <MemoryRouter>
             <App />
            </MemoryRouter>
          </Provider>);
});

test('matches snapshot', () => {
  const {useFragment} = render( <Provider store={store}>
                                  <MemoryRouter>
                                    <App />
                                    </MemoryRouter>
                                  </Provider>
                                );
  expect(useFragment).toMatchSnapshot();
});





