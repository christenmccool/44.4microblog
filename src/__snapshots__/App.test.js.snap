// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API is called 1`] = `undefined`;

exports[`matches snapshot 1`] = `undefined`;
